packages/bsky/tests/views/threadgating.test.ts

```diff
@@ -242,8 +247,72 @@ describe('views with thread gating', () => {
     expect(reply.post.uri).toEqual(aliceReply.ref.uriStr)
   })
 
+  it('applies gate for follower rule.', async () => {
+    const post = await sc.post(sc.dids.carol, 'follower rule')
+    await pdsAgent.api.app.bsky.feed.threadgate.create(
+      { repo: sc.dids.carol, rkey: post.ref.uri.rkey },
+      {
+        post: post.ref.uriStr,
+        createdAt: iso(),
+        allow: [{ $type: 'app.bsky.feed.threadgate#followerRule' }],
+      },
+      sc.getHeaders(sc.dids.carol),
+    )
+    await network.processAll()
+
+    // dan does not follow carol, can't reply
+    await sc.reply(
+      sc.dids.dan,
+      post.ref,
+      post.ref,
+      'follower rule reply disallow',
+    )
+
+    // alice follows carol, can reply
+    const aliceReply = await sc.reply(
+      sc.dids.alice,
+      post.ref,
+      post.ref,
+      'follower rule reply allow',
+    )
+    await network.processAll()
+    const {
+      data: { thread: danThread },
+    } = await agent.api.app.bsky.feed.getPostThread(
+      { uri: post.ref.uriStr },
+      {
+        headers: await network.serviceHeaders(
+          sc.dids.dan,
+          ids.AppBskyFeedGetPostThread,
+        ),
+      },
+    )
+    assert(isThreadViewPost(danThread))
+    expect(danThread.post.viewer?.replyDisabled).toBe(true)
```

Comment:
For sake of completeness, might as well add a check to confirm that Dan's reply wasn't accepted and included in the thread response

Reply:
Ok I see, it's implicitly checked with the length check there. That's what I mean by these tests being kinda messy 😛 all good for now, thanks for the explainer 👍